src/DatabaseCommunicator/QueryBuilder.chpl

```diff
@@ -0,0 +1,126 @@
+module QueryBuilder {
+  use DatabaseCommunicator.DatabaseCommunicationObjects.ErrorTypes;
+
+  /*** Query Building ***/
+
+  /* 
+  This record helps in the creation of SQL queries and statements,
+  something like `PreparedStatement` used by the JDBC in Java.
+  Useful for ensuring proper data types and prevention of SQL injection.
+
+  For example: `SELECT id FROM USERS WHERE name = ?1 and address = ?2`
+  The `?1` and `?2` placeholders can be replaced with setValue.
+  Note that there must be atleast one space before
+  the placeholder to be recognized as one.
+  Hence, for `SELECT id FROM USERS WHERE name =?1 and address = ?2`,
+  `?1` will not be recognized as a placeholder and `?2` will be.
+  */
+  class Statement {
+    var _statementUnformatted: string;
+    var _toSubstitute: bool;
+    var _finalStatement: string;
+    var _placeholderRemains: bool;
+
+    /*
+    Initialize an SQL statement.
+      :arg statement: The SQL statement
+      :type statement: string
+      :arg toSubstitue: True if the arg `statement` contains placeholder question marks to be substituted
+      :type toSubstitue: bool
+    */
+    
+    proc init(statement: string, toSubstitute: bool = false) {

```

Comment:
I think there should be another constructor for these cases.
```chpl
var st3: Statement = new Statement("SELECT * FROM CONTACTS WHERE id = ?1 AND name = ?2 AND has_legs = ?3;", true, 1, 'testUser', true);
```

Reply:
There's slight ambiguity that can be caused by this kind of a constructor in my opinion. Consider the following cases:
1. The placeholders do not appear in serial order in the statement. So if we have something like:
```sql
SELECT * FROM CONTACTS WHERE id = ?3 AND name = ?1 AND has_legs = ?2;
```
in which order is the user expected to give the variadic arguments? Should it be in the order the placeholders appear in the statement or in natural order?

2. The placeholders are not in order and skip numbers in the index. So something like this is still valid, but it will be quite ambiguous as to which argument sets which placeholder if we use a constructor with variadic arguments:
```sql
SELECT * FROM CONTACTS WHERE id = ?3 AND name = ?1;
```

I think we can have a default behaviour for such cases if we do decide to go for the constructor with variadic args though. What are your thoughts about this?

Reply:
Just for clarification regarding the second point, would something like this be considered valid?:

```chapel
var st3: Statement = new Statement("SELECT * FROM CONTACTS WHERE id = ?3 AND name = ?1 AND parmanent_address = ?2 AND current_address = ?2 AND future_address = ?4;", true, "testUser", "Lux", 1);
st3.setValue(4, "home");
```

In other words, the user has to set all the placeholders in the constructor or not set any at all (that's the other constructor), right?

Reply:
I think it's okay to make it valid as users may want to modify the placeholder values later on as well.